### PR TITLE
testing: remove recursive make for testing

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

Recursive make does not observe full dependency, result in a `testing` with outdated binary. This PR attempt to consolidate recursive make into a single Makefile.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by hzhou** [comment]
test: yaksa

**Comment by hzhou** [comment]
test:yaksa/quick

**Comment by hzhou** [comment]
test: yaksa/quick

**Comment by hzhou** [comment]
test: yaksa/quick

**Comment by hzhou** [comment]
test:yaksa

**Comment by hzhou** [comment]
test:yaksa/quick

**Comment by hzhou** [comment]
test:yaksa/quick

File: test/runtests.py [hunk 1]
```diff
@@ -187,11 +187,11 @@ def run_testlist(testlist):
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument('--testlist', help='testlist file to execute', required=True)
+    parser.add_argument('testlists', help='testlist files to execute', nargs='+')
```
Comment (pavanbalaji) [inline test/runtests.py:190]: This should be a required argument.

File: maint/gentests.py [hunk 2]
```diff
@@ -25,15 +26,17 @@ def create_testlist(testlist):
...
         outfile = open(testlist, "w")
     except:
         sys.stderr.write("error creating testlist %s\n" % testlist)
         sys.exit()
 
     sys.stdout.write("generating simple tests ... ")
-    outfile.write("simple_test\n")
-    outfile.write("threaded_test\n")
+    outfile.write(prefix + "/" + "simple_test\n")
+    outfile.write(prefix + "/" + "threaded_test\n")
```
Comment (pavanbalaji) [inline maint/gentests.py:39]: This should use `os.path.join`.

File: maint/gentests.py [hunk 3]
```diff
@@ -74,7 +79,7 @@ def gen_pack_iov_tests(fn, testlist, create):
                         else:
                             iters = 128
 
-                        outstr = fn + " "
+                        outstr = prefix + "/" + fn + " "
```
Comment (pavanbalaji) [inline maint/gentests.py:82]: This should use `os.path.join`.

File: maint/gentests.py [hunk 4]
```diff
@@ -108,7 +115,7 @@ def gen_flatten_tests(testlist):
             else:
                 iters = 128
 
-            outstr = "flatten "
+            outstr = prefix + "/" + "flatten "
```
Comment (pavanbalaji) [inline maint/gentests.py:118]: This should use `os.path.join`

File: maint/gentests.py (hunk 2 above)
Comment (hzhou) [reply maint/gentests.py]: fixed.

File: maint/gentests.py (hunk 3 above)
Comment (hzhou) [reply maint/gentests.py]: fixed.

File: maint/gentests.py (hunk 4 above)
Comment (hzhou) [reply maint/gentests.py]: fixed.

File: test/runtests.py (hunk 1 above)
Comment (hzhou) [reply test/runtests.py:190]: It is. `nargs='+'` means require 1 or more.

**Comment by hzhou** [comment]
test:yaksa

**Pull request merged**